src/test/java/uk/co/ramp/people/PersonTest.java

```diff
@@ -0,0 +1,266 @@
+        Assert.assertEquals(RECOVERED, person.getStatus());
+    }
+
+    @Test
+    public void testExposedBy() {
+
+        Assert.assertEquals(SUSCEPTIBLE, person.getStatus());
+        Assert.assertEquals(-1, person.getExposedBy());
+        int exposer = rnd.nextInt(0, 1000);
+        person.updateStatus(EXPOSED, 0, exposer);
+
+        Assert.assertEquals(EXPOSED, person.getStatus());
+        Assert.assertEquals(exposer, person.getExposedBy());
+
+    }
+
+
+    @Test
+    public void getNextStatusChange() {
+        Assert.assertEquals(SUSCEPTIBLE, person.getStatus());
+        Assert.assertEquals(-1, person.getExposedBy());
+        int exposer = rnd.nextInt(0, 1000);
+        int timeNow = rnd.nextInt(0, 100);
+        person.updateStatus(EXPOSED, timeNow, exposer);
+
+        Assert.assertEquals(EXPOSED, person.getStatus());
+        Assert.assertTrue(person.getNextStatusChange() > timeNow);
+
+    }
+
+    @Test(expected = InvalidStatusTransitionException.class)
+    public void checkTime() throws NoSuchFieldException {
+
+        int time = rnd.nextInt(0, 100);
+        int flatTransition = (int) diseaseProperties.getMeanTimeToInfected();
+
+        person.updateStatus(EXPOSED, time - flatTransition);
+
+        person.checkTime(time);
+
+        flatTransition = (int) diseaseProperties.getMeanTimeToRecovered();
+
+        Assert.assertEquals(INFECTED, person.getStatus());
+        Assert.assertEquals(time + flatTransition, person.getNextStatusChange());
+
+
+        time += flatTransition;
+        person.checkTime(time);
+        Assert.assertEquals(RECOVERED, person.getStatus());
+        Assert.assertEquals(-1, person.getNextStatusChange());
+
+        person.updateStatus(SUSCEPTIBLE, ++time);
+        FieldSetter.setField(person, person.getClass().getDeclaredField("nextStatusChange"), time);
+
+        try {
+            person.checkTime(time);
+            Assert.fail();
+        } catch (InvalidStatusTransitionException e) {
+            Assert.assertThat(appender.getOutput(), containsString(String.format("Changing status from RECOVERED for person.id %d is not a valid transition", id)));
+            throw e;
+        }
+    }
+
+    @Test
+    public void notNow() {
+        int time = rnd.nextInt(0, 100);
+        int flatTransition = (int) diseaseProperties.getMeanTimeToInfected();
+
+        person.updateStatus(EXPOSED, time - flatTransition);
+        person.checkTime(time - 10);
+
+        // event times don't match, so nothing happens
+        Assert.assertEquals("", appender.getOutput());
+    }
+
+    @Test
+    public void randomExposure() {
+        int time = rnd.nextInt(0, 1000);
+        person.randomExposure(time);
+        Assert.assertThat(appender.getOutput(), containsString(String.format("Person with id: %d has been randomly exposed at time %d", id, time)));
+    }
+
+
+    @Test(expected = InvalidStatusTransitionException.class)
+    public void invalidExposure() {
+        int time = rnd.nextInt(0, 1000);
```

Comment:
This highlights an interesting design choice:  A person who has already been exposed cannot be exposed again.  (I would have guessed that anyone could be exposed, it just wouldn't effect some people.)

Reply:
I may be wrong, but from the SEIR papers I've read, the 'exposed' state is more like a pre-symptomatic, infected but not infectious state. Happy to be corrected.